### PR TITLE
Raise nicer error message if factor is not integer

### PR DESCRIPTION
**Pull request by larrybradley**
### Description

With this PR, ``discretize_model`` will now raise an informative ``ValueError`` if ``mode='oversample'`` and ``factor`` does not have an integer value.

### CODE REVIEW

**Comment by pllim** [comment]
Remote data passed here. Does this mean we no longer need #14789 ? 🤯

**Review by keflavich — APPROVED** [review]
looks good, I just suggest including the type requirement in the documentation too

File: astropy/convolution/utils.py [hunk 1]
```diff
@@ -120,7 +120,8 @@ def discretize_model(model, x_range, y_range=None, mode="center", factor=10):
                 bins. For 2D models, the interpolation is bilinear.
             * ``'oversample'``
                 Discretize model by taking the average of model values
-                on an oversampled grid.
+                in the pixel bins on an oversampled grid. Use the
+                ``factor`` keyword to set the oversampling factor.
```
Comment (keflavich) [inline astropy/convolution/utils.py:124]: maybe
```suggestion
                ``factor`` keyword to set the (integer) oversampling factor.
```

**Review by keflavich — APPROVED** [review]
looks good, I just suggest including the type requirement in the documentation too

File: astropy/convolution/utils.py (hunk 1 above)
Comment (larrybradley) [reply astropy/convolution/utils.py]: Done, thanks!  I also explicitly added that in the `factor` description.

**Pull request merged**